timeout_sampler/__init__.py

```diff
@@ -228,3 +231,30 @@ def remaining_time(self) -> float:
         Return the remaining part of timeout since the object was created.
         """
         return self.start_time + self.timeout - time.time()
+
+
+def timeout_sampler_deco(
+    wait_timeout: int,
+    sleep: int,
+    exceptions_dict: dict[type[Exception], list[str]] | None = None,
```

Comment:
Would you consider a rename to reflect better what they actually represent?
- wait_timeout ==> overall_timeout , max_timeout or just a simple "timeout".
- sleep ==> interval
- exceptions_dict ==> expected_exceptions